src/plugins/wizard/public/application/utils/wizard_state_validation.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export const wizardStateValidation = (wizardState) => {
+  /* eslint-disable */
+  const Ajv = require('ajv');
+  const ajv = new Ajv();
+  const vizTypeEnum = ['metric', 'line', 'area', 'histogram'];
```

Comment:
Change the type here as string because we don't need to be too strict with the type check since user could install a plugin that introduces a new type.